core/tx_pool.go

```diff
@@ -707,8 +719,15 @@ func (pool *TxPool) validateTx(tx types.PoolTransaction, local bool) error {
 		}
 		return ErrInvalidSender
 	}
+
+	inAllowedTxs := false
+	if allowedTx, exists := pool.config.AllowedTxs[from]; exists {
+		to := tx.To()
+		inAllowedTxs = to != nil && *to == allowedTx.To && bytes.Equal(tx.Data(), allowedTx.Data)
+	}
+
 	// Make sure transaction does not have blacklisted addresses
-	if _, exists := (pool.config.Blacklist)[from]; exists {
+	if _, exists := (pool.config.Blacklist)[from]; exists && !inAllowedTxs {
 		if b32, err := hmyCommon.AddressToBech32(from); err == nil {
 			return errors.WithMessagef(ErrBlacklistFrom, "transaction sender is %s", b32)
 		}
```

Comment:
nitpick: Can you change the error string of `ErrBlacklistFrom` and print the `data` and `to` values in the message for better logging?